### DISCUSSION TITLE
The ownership of the DecentralizedStableCoin.sol ....

### DISCUSSION THREAD

**Discussion by Devshenoyprasad** [Cyfrin/foundry-full-course-cu#167] [Q&A] [answered]
I was going through the Stablecoin Defi project. I noticed that that the DecentralizedStableCoin is a ownable contract that is to be owned by the DSCEngine.sol ant the function in the contract i.e burn and mint have onlyOwner modifier. I am trying to understand how DSCEngine contract became the owner of DecentralizedStableCoin ??

**Comment by cromewar** [comment, marked as answer]
Hello @Devshenoyprasad 
Well that is a really good question, let me try to answer it:
This ownership assignment typically happens at the time of deployment or shortly after, when a function is called to set the owner. The contract deployer would call a function to set DSCEngine as the owner of the DecentralizedStableCoin. This function would likely be protected by the onlyOwner modifier to ensure that only the deployer (initial owner) can change the ownership to DSCEngine.
While this design gives DSCEngine a certain level of control over the DecentralizedStableCoin, it's important to note that this isn't truly decentralized. A truly decentralized stablecoin wouldn't have a single owner entity that can unilaterally mint or burn tokens. Instead, these critical functions would be controlled through consensus mechanisms such as a DAO (Decentralized Autonomous Organization) where token holders vote on proposals, including minting or burning of tokens.

Let me know if you need any extra help.

**Reply by Devshenoyprasad** [reply]
Thank you for the answer. Can u just point out where exactly in the cote the tranfership of owner takes place?

**Reply by ghost** [reply]
On [line 28](https://github.com/Cyfrin/foundry-defi-stablecoin-f23/blob/main/script/DeployDSC.s.sol#L28) of the `DeployDSC.s.sol`

**Reply by Devshenoyprasad** [reply]
Thank you for pointing it out

**Reply by ghost** [reply]
You're welcome, don't forget to mark cromewar's reply as the answer

**Comment by Devshenoyprasad** [comment]
How about making the DSCEngine.sol to deploy DecentralizedStableCoin.sol rather that passing the contract address as a parameter to the constructor of DSCEngine.sol?? It will then make DSCEngine the owner of DecentralizedStableCoin i.e the ERC-20 contract.
